packages/amplify_auth_cognito/lib/src/CognitoSession/AWSCredentials.dart

```diff
@@ -19,14 +19,12 @@ import 'package:amplify_auth_plugin_interface/amplify_auth_plugin_interface.dart
 import 'package:flutter/material.dart';
 
 class AWSCredentials {
-  String awsAccessKey;
-  String awsSecretKey;
-  String sessionToken;
-  AWSCredentials.init({@required LinkedHashMap<dynamic, dynamic> creds}) {
-    if (creds != null) {
-      this.awsAccessKey = creds.containsKey("awsAccessKey") ? creds["awsAccessKey"] : null;
-      this.awsSecretKey = creds.containsKey("awsSecretKey") ? creds["awsSecretKey"] : null;
-      this.sessionToken = creds.containsKey("sessionToken") ? creds["sessionToken"] : null;
-    }
-  }
+  String? awsAccessKey;
+  String? awsSecretKey;
+  String? sessionToken;
```

Comment:
@haverchuck should these fields be non nullable?

Reply:
Similar to the `AuthCodeDeliveryDetails` answer above - the `AWSCredentials` and `AWSCognitoUserPoolTokens` could themselves be made nullable, and then if present the indiviudal values within would not be nullable. This would require some changes at the platform level, though fairly minimal.

Reply:
As noted earlier, this entire API should be refactored, because the `AWSCredentials` and `AWSCognitoUserPoolTokens` should be able to throw exceptions when accessed in particular cases. This change would be a good first step.

Reply:
Cool the fields of `AWSCredentials` and `AWSCognitoUserPoolTokens` have all been changed to nullable

Reply:
I believe @haverchuck mentioned to make the fields inside these objects non-nullable but the objects themselves nulalble in the container object.

Reply:
Yes thanks for that clarification Praveen.  We are discussing now to resolve this.